### PR TITLE
chore: improve password error message

### PR DESCRIPTION
**Pull request by fundon**
chore: improve error message

chore: add password minlength & maxlength i18n

chore: check max length

fix: i18n variables

feat: add CredentialsRequirementType

### CODE REVIEW

**Comment by fundon** [comment]
* **#6255** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/6255?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @fundon and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/backend/server/src/core/config.ts [hunk 1]
```diff
@@ -22,6 +22,21 @@ export function ADD_ENABLED_FEATURES(feature: ServerFeature) {
...
+export class PasswordLimitsType {
+  @Field()
+  minLength!: number;
+
+  @Field()
+  maxLength!: number;
+}
+
+@ObjectType()
+export class AuthPrivilegeType {
```
Comment (EYHN) [inline packages/backend/server/src/core/config.ts]:
```suggestion
export class AuthPolicyType {
```

I think to express overall authorization rules or policies, choosing `AuthPolicy` might be more appropriate. Whereas if you need to specifically describe the permissions or privileges of a particular user or entity, then you should use `AuthPrivilege`

File: packages/frontend/component/src/components/auth-components/password-input/index.tsx [hunk 2]
```diff
@@ -27,30 +30,37 @@ export const PasswordInput: FC<
...
+        case 2:
+          setStatus('medium');
+          break;
+        case 3:
+          setStatus('strong');
+          break;
+      }
+    },
+    [passwordLimits]
+  );
```
Comment (darkskygit) [inline packages/frontend/component/src/components/auth-components/password-input/index.tsx]: can we write a zod type validator to simplify this logic? its too long

File: packages/backend/server/src/core/config.ts [hunk 3]
```diff
@@ -22,6 +22,21 @@ export function ADD_ENABLED_FEATURES(feature: ServerFeature) {
...
+export class PasswordLimitsType {
+  @Field()
+  minLength!: number;
+
+  @Field()
+  maxLength!: number;
+}
+
+@ObjectType()
+export class AuthPolicyType {
```
Comment (forehalo) [inline packages/backend/server/src/core/config.ts]: `CredentialsRequirementType`, `Policy` term might be used in further Permission module

File: packages/frontend/component/src/components/auth-components/password-input/index.tsx [hunk 4]
```diff
@@ -1,104 +1,200 @@
...
+        ctx.addIssue({
+          code: ZodIssueCode.custom,
+          message: t['com.affine.auth.set.password.message.maxlength']({
+            max: String(maxLength),
+          }),
+          params: {
+            status: 'maximum',
+          },
+        });
+      }
```
Comment (darkskygit) [inline packages/frontend/component/src/components/auth-components/password-input/index.tsx]: why not use min().max() directly? error message can process in view after, and we can use error code `too_small` and `too_big` to identify the error:
image

Comment (fundon) [reply packages/frontend/component/src/components/auth-components/password-input/index.tsx]: We want to pass the status parameter, so use `ZodCustomIssue`.

File: packages/frontend/component/src/components/auth-components/password-input/index.tsx [hunk 5]
```diff
@@ -1,104 +1,200 @@
...
+          value: 'medium',
+          minDiversity: 4,
+          minLength: 8,
+        },
+        {
+          id: 2,
+          value: 'strong',
+          minDiversity: 4,
+          minLength: 10,
+        },
```
Comment (darkskygit) [inline packages/frontend/component/src/components/auth-components/password-input/index.tsx]: extract as constant

File: packages/frontend/component/src/components/auth-components/password-input/index.tsx (hunk 4 above)
Comment (darkskygit) [reply packages/frontend/component/src/components/auth-components/password-input/index.tsx]: where we use the message? i see this i18n call already in the view

Comment (darkskygit) [reply packages/frontend/component/src/components/auth-components/password-input/index.tsx]: where we use the message? i see this i18n call already in the view

Comment (fundon) [reply packages/frontend/component/src/components/auth-components/password-input/index.tsx]: Good catch. 
Removed.

**Review by forehalo — APPROVED** [review]

**Review by EYHN — APPROVED** [review]

**Pull request merged**